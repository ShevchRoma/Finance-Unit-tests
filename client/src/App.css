/*////////////Normalize///////////*/
* {
  padding: 0;
  margin: 0;
  border: 0;
  list-style: none;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

ul,
ol,
li {
  list-style: none;
}

a {
  text-decoration: none;
  outline: none;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-weight: inherit;
  font-size: inherit;
}

img {
  vertical-align: top;
}
body {
  height: 100%;
  line-height: 1;
  font-size: 16px;
  font-weight: 700;
  font-family: "Open Sans";
  scroll-behavior: smooth;
  text-rendering: optimizeSpeed;
}
.wrapper {
  max-width: 1440px;
  overflow: hidden;
}
.container {
  max-width: 1280px;
  margin: 0 auto;
}
/*//////////Styles/////////*/

.app__top {
  margin-bottom: 80px;
  padding-top: 20px;
}

.app__bottom {
  margin-bottom: 25px;
}

.bottom__body {
  display: flex;
  flex-direction: column;
  border-radius: 12px;
  padding: 20px;
  margin: 0 auto;
  max-width: 650px;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
}

button {
  border: 1px solid rgb(239, 236, 236);
  background: rgb(214, 213, 213);
  padding: 15px 0;
  margin: 0 auto;
  display: flex;
  justify-content: center;
  width: 350px;
  color: #fff;
  border-radius: 15px;
  text-transform: uppercase;
  font-weight: bold;
  transition: 0.2s ease;
  background: linear-gradient(
    10deg,
    rgb(225, 228, 234) 0%,
    rgb(190, 192, 197) 50%,
    rgb(61, 123, 238) 50%,
    rgb(196, 196, 199) 90%
  );
  text-shadow: -1px -1px 0 hsl(60, 2%, 79%), -2px -2px 1px hsl(60, 1%, 34%);
}

button:hover {
  text-shadow: -1px -1px 0 hsl(0, 0%, 98%), -1px -1px 1px hsl(60, 16%, 96%);
  color: rgb(67, 132, 236);
  box-shadow: rgba(53, 79, 245, 0.35) 0px 5px 15px;
  background: linear-gradient(
    150deg,
    rgb(225, 228, 234) 0%,
    rgb(190, 192, 197) 50%,
    rgb(190, 192, 197) 50%,
    rgb(141, 142, 145) 100%
  );
}
