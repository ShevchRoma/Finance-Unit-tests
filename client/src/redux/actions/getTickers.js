import { socket } from "../../api/socketApi";
import {
  GET_TICKERS,
  GET_TICKERS_ERROR,
  GET_TICKERS_SUCCESS,
} from "../reducers/tickersReducer";

export const getTickers = () => async (dispatch) => {
  try {
    dispatch({
      type: GET_TICKERS,
    });
    socket.emit("start");
    socket.on("ticker", (tickers) => {
      dispatch({
        type: GET_TICKERS_SUCCESS,
        payload: tickers,
      });
    });
  } catch (err) {
    dispatch({
      type: GET_TICKERS_ERROR,
      payload: err.message,
    });
  }
};
