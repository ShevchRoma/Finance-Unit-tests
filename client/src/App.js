import React from "react";
import "./App.css";
import { useSelector, useDispatch } from "react-redux";
import { getTickers } from "./redux/actions/getTickers";
import Ticker from "./components/Ticker/Ticker";
import Time from "./components/Time/Time";

function App() {
  const { tickers } = useSelector((state) => state.tickers);

  const dispatch = useDispatch();

  /* or...
    React.useEffect(() =>{
       dispatch(getTickers())
    },[dispatch]) */

  return (
    <div className="wrapper">
      <div className="app">
        <div className="app__body">
          <div className="app__top top">
            <div className="container">
              <Time />
            </div>
          </div>
          <div className="app__bottom bottom">
            <div className="container">
              {!tickers ? (
                <button onClick={() => dispatch(getTickers())}>
                  Click Me To start
                </button>
              ) : (
                <div className="bottom__body">
                  <div className="bottom__row">
                    {tickers &&
                      tickers.map((item) => (
                        <Ticker item={item} key={item.ticker} />
                      ))}
                  </div>
                </div>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default App;
