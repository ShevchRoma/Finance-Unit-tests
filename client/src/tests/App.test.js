import { render, screen, fireEvent } from "@testing-library/react";
import App from "../App";
import { useSelector } from "react-redux";
import React from "react";
import * as reduxHooks from "react-redux";
import * as actions from "../redux/actions/getTickers";

jest.mock("react-redux");
jest.mock("React", () => ({
  ...jest.requireActual("React"),
  useEffect: jest.fn(),
}));

const mockedDispatch = jest.spyOn(reduxHooks, "useDispatch");

const tickers = [
  {
    ticker: "AAPL",
    exchange: "NASDAQ",
    price: "128.56",
    change: "137.32",
    change_percent: "0.51",
    dividend: "0.20",
    yield: "0.58",
  },
  {
    ticker: "GOOGL",
    exchange: "NASDAQ",
    price: "237.08",
    change: "154.38",
    change_percent: "0.10",
    dividend: "0.18",
    yield: "0.98",
  },
];

describe("App", () => {
  it("should create App", () => {
    mockedDispatch.mockResolvedValue(jest.fn());
    useSelector.mockReturnValue([]);

    const component = render(<App />);

    expect(component).toMatchSnapshot();
  });
  it("should create App  with tickers items", () => {
    useSelector.mockReturnValue(tickers);

    const component = render(<App />);

    expect(component).toMatchSnapshot();
  });

  it("should dispatch actions", () => {
    const dispatch = jest.fn();

    mockedDispatch.mockReturnValue(dispatch);
    useSelector.mockReturnValue([]);
    const mockedGetTickers = jest.spyOn(actions, "getTickers");
    render(<App />);
    fireEvent.click(screen.getByRole("button"));

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(mockedGetTickers).toHaveBeenCalled();
  });
});
