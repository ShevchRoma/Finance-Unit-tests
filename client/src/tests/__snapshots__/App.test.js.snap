// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should create App  with tickers items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="app"
        >
          <div
            class="app__body"
          >
            <div
              class="app__top top"
            >
              <div
                class="container"
              >
                <div
                  class="top__body"
                >
                  <div
                    class="top__column column-top"
                  >
                    <div
                      class="column-top__row"
                    >
                      <div
                        class="column-top__dot"
                      />
                      <p
                        class="column-top__name"
                      >
                        15:40
                      </p>
                    </div>
                    <div
                      class="column-top__row"
                    >
                      <div
                        class="column-top__percent"
                      >
                        Time
                      </div>
                      <div
                        class="column-top__coefficient"
                      >
                        Decor
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="app__bottom bottom"
            >
              <div
                class="container"
              >
                <button>
                  Click Me To start
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="app"
      >
        <div
          class="app__body"
        >
          <div
            class="app__top top"
          >
            <div
              class="container"
            >
              <div
                class="top__body"
              >
                <div
                  class="top__column column-top"
                >
                  <div
                    class="column-top__row"
                  >
                    <div
                      class="column-top__dot"
                    />
                    <p
                      class="column-top__name"
                    >
                      15:40
                    </p>
                  </div>
                  <div
                    class="column-top__row"
                  >
                    <div
                      class="column-top__percent"
                    >
                      Time
                    </div>
                    <div
                      class="column-top__coefficient"
                    >
                      Decor
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="app__bottom bottom"
          >
            <div
              class="container"
            >
              <button>
                Click Me To start
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App should create App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="app"
        >
          <div
            class="app__body"
          >
            <div
              class="app__top top"
            >
              <div
                class="container"
              >
                <div
                  class="top__body"
                >
                  <div
                    class="top__column column-top"
                  >
                    <div
                      class="column-top__row"
                    >
                      <div
                        class="column-top__dot"
                      />
                      <p
                        class="column-top__name"
                      >
                        15:40
                      </p>
                    </div>
                    <div
                      class="column-top__row"
                    >
                      <div
                        class="column-top__percent"
                      >
                        Time
                      </div>
                      <div
                        class="column-top__coefficient"
                      >
                        Decor
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="app__bottom bottom"
            >
              <div
                class="container"
              >
                <button>
                  Click Me To start
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="app"
      >
        <div
          class="app__body"
        >
          <div
            class="app__top top"
          >
            <div
              class="container"
            >
              <div
                class="top__body"
              >
                <div
                  class="top__column column-top"
                >
                  <div
                    class="column-top__row"
                  >
                    <div
                      class="column-top__dot"
                    />
                    <p
                      class="column-top__name"
                    >
                      15:40
                    </p>
                  </div>
                  <div
                    class="column-top__row"
                  >
                    <div
                      class="column-top__percent"
                    >
                      Time
                    </div>
                    <div
                      class="column-top__coefficient"
                    >
                      Decor
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="app__bottom bottom"
          >
            <div
              class="container"
            >
              <button>
                Click Me To start
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
